app/models/team_repository.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class TeamRepository < ApplicationRecord
+  enum permission_level: Extends::SHARED_INVENTORIES_PERMISSION_LEVELS
+
+  belongs_to :team
+  belongs_to :repository
+
+  validates :permission_level, presence: true
```

Comment:
Maybe it is worth also create scope validation, to check that there is only one team<=>repository pair